modules/SystemsBiology.py

```diff
@@ -0,0 +1,164 @@
+                            description = SYSBIO_PMID_MAP[pmid]['method'],
+                            reference = "http://europepmc.org/abstract/MED/{0}".format(pmid)
+                        ),
+                        # TODO: add scores as p-value or a rank-based score (to be decided)
+                        # Default score is 0.5 if no scores are provided
+                        value=0.5
+                    )
+
+                    evidenceString = opentargets.Literature_Curated()
+                    evidenceString.validated_against_schema_version = Config.VALIDATED_AGAINST_SCHEMA_VERSION
+                    evidenceString.access_level = "public"
+                    evidenceString.type = "affected_pathway"
+                    evidenceString.sourceID = "sysbio"
+
+                    # *** Build unique_association_field object ***
+                    evidenceString.unique_association_fields = {}
+                    evidenceString.unique_association_fields['pmid'] = pmid
+                    evidenceString.unique_association_fields['gene_set'] = gene_set_name
+                    evidenceString.unique_association_fields['gene_name'] = target_name
+
+                    # *** Build target object ***
+                    if target_name in self.symbols:
+                        ensembl_gene_id = self.symbols[target_name]
+
+                        evidenceString.target = bioentity.Target(
+                            id="http://identifiers.org/ensembl/{0}".format(ensembl_gene_id),
+                            target_name=target_name,
+                            # TODO activity is a required field in target object, currently set as unknown
+                            activity="http://identifiers.org/cttv.activity/unknown",
+                            target_type='http://identifiers.org/cttv.target/gene_evidence'
+                        )
+                        # *** Build disease object ***
+                        evidenceString.disease = bioentity.Disease(
+                            id=disease_id,
+                            name=disease_name
+                        )
+                        # *** Build evidence oject ***
+                        evidenceString.evidence = evidence_core.Literature_Curated()
+                        evidenceString.evidence.date_asserted = now.isoformat()
+                        evidenceString.evidence.is_associated = True
+                        evidenceString.evidence.evidence_codes = ["http://purl.obolibrary.org/obo/ECO_0000053"]
+                        evidenceString.evidence.provenance_type = provenance_type
+                        evidenceString.evidence.resource_score = resource_score
+
+
+                        error = evidenceString.validate(logging)
+
+                        #print(evidenceString.to_JSON(indentation=None))
+                        ##TODO issue with append, take only last item of the gene
+                        self.evidence_strings.append(evidenceString)
```

Comment:
@AsierGonzalez this limit your process to a small number of evidences not greater than your local machine memory. A proper approach is using an each line process and saving each line as all lines are independent. have a look at the package [pyfunctional](https://pypi.org/project/PyFunctional/) to solve this problem.. Also, it is always good having also the package [more-itertools](https://pypi.org/project/more-itertools/)